DevProxy.Plugins/Reporting/UrlDiscoveryPlugin.cs

```diff
@@ -28,19 +29,18 @@ public override Task AfterRecordingStopAsync(RecordingArgs e, CancellationToken
         }
 
         var requestLogs = e.RequestLogs
-            .Where(l => ProxyUtils.MatchesUrlToWatch(UrlsToWatch, l.Context?.Session.HttpClient.Request.RequestUri.AbsoluteUri ?? ""));
+            .Where(l => ProxyUtils.MatchesUrlToWatch(UrlsToWatch, l.Request!.RequestUri!.AbsoluteUri ?? ""));

```

Comment:
The `!.` looks a bit messy. If it can be null, we need to do more robust checks to avoid runtime failures.